flink-ml-core/src/main/java/org/apache/flink/ml/common/iteration/TerminateOnMaxIter.java

```diff
@@ -27,16 +27,17 @@
  * threshold.
  *
  * <p>When the output of this FlatMapFunction is used as the termination criteria of an iteration
- * body, the iteration will be executed for at most the given `numRounds` rounds.
+ * body, the iteration will be executed for at most the given `maxIter` rounds.
  *
  * @param <T> The class type of the input element.
  */
-public class TerminateOnMaxIterationNum<T>
+public class TerminateOnMaxIter<T>
```

Comment:
I would stick with TerminateOnMaxIter because `RoundBasedTerminationCriteria ` is not in the source code. 

Moreover, it is not consistent with the `TerminateOnMaxIter ` in the following aspects:
(1) It does not accept a generic as input
(2) It does not run the same number of iterations as `TerminateOnMaxIter`
(3) It does not specify a tolerance based termination criteria

Reply:
Can we remove the `RoundBasedTerminationCriteria` entirely, and use `TerminateOnMaxIter` in the test code instead?

"replace A with B" means that we can remove `A` and use `B` where `A` was needed.